.github/workflows/build-notebooks-TEMPLATE.yaml

```diff
@@ -96,14 +104,38 @@ jobs:
           mkdir -p $HOME/.local/share/containers/storage/tmp
 
       # https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#push
-      - name: "push: make ${{ inputs.target }}"
+      - name: "push|schedule: make ${{ inputs.target }}"
         run: "make ${{ inputs.target }}"
-        if: "${{ fromJson(inputs.github).event_name == 'push' }}"
+        if: ${{ fromJson(inputs.github).event_name == 'push' || fromJson(inputs.github).event_name == 'schedule' }}
         env:
           IMAGE_TAG: "${{ github.ref_name }}_${{ github.sha }}"
           IMAGE_REGISTRY: "ghcr.io/${{ github.repository }}/workbench-images"
           CONTAINER_BUILD_CACHE_ARGS: "--cache-from ${{ env.CACHE }} --cache-to ${{ env.CACHE }}"
 
+      - name: "schedule: run Trivy vulnerability scanner"
+        if: "${{ fromJson(inputs.github).event_name == 'schedule' }}"
+        run: |
+          TRIVY_VERSION=0.52.2
+          REPORT_FOLDER=${{ github.workspace }}/report
+          REPORT_FILE=trivy-report.md
+          REPORT_TEMPLATE=trivy-markdown.tpl
+
+          mkdir -p $REPORT_FOLDER
+          cp ci/$REPORT_TEMPLATE $REPORT_FOLDER
+          IMAGE_NAME=$(podman images --format '{{.Repository}}:{{.Tag}}' --no-trunc | head -n 1)
```

Comment:
I had another look at this and I suspect that `IMAGE_NAME` may not be the image we intend to scan. It would be best to construct IMAGE_NAME from the variables the step above uses.

```

          IMAGE_TAG: "${{ github.ref_name }}_${{ github.sha }}"
          IMAGE_REGISTRY: "ghcr.io/${{ github.repository }}/workbench-images"
```

Can you add a debug print, so we are sure what we're scanning?

```suggestion
          IMAGE_NAME=$(podman images --format '{{.Repository}}:{{.Tag}}' --no-trunc | head -n 1)
          echo "[INFO] IMAGE_NAME=${IMAGE_NAME}"
```

Reply:
I had a problem when I tried to use the variables but let me double-check.

Reply:
just the echo would be sufficient for now, just that I can always check what's it actually scanning

Reply:
Just tested. It was scanning the last build-cache created, which had the same ImageId of the image we wanted to check. Anyway, I'll change to:
```bash
IMAGE_NAME=ghcr.io/${{ github.repository }}/workbench-images:${{ inputs.target }}-${{ github.ref_name }}_${{ github.sha }}
echo "Scanning $IMAGE_NAME"
```
Good catch!

Reply:
The only thing is that it becomes too coupled with the logic defined in the Makefile.

Reply:
github does not have sad emoji, and I can't use similarly looking confused emoji, because I'm not actually confused

😢